lib/acts_as_taggable_on_mongoid/taggable/utils/tag_list_diff.rb

```diff
@@ -0,0 +1,94 @@
+# frozen_string_literal: true
+
+module ActsAsTaggableOnMongoid
+  module Taggable
+    module Utils
+      # A utility class for creating new taggings and deleting old taggings when the list of tags for an object
+      # is changed and that change is saved.
+      #
+      # The class will respect the preserve_tag_order? option for the tag and will delete any existing taggings
+      # and recreate them from the first moved tag onwards.
+      #
+      # If order is not preserved, it will simply delete unused taggings and add new taggings that do not already
+      # exist.
+      class TagListDiff
```

Comment:
Everything in this class had been one big ugly function in ActsAsTaggableOn.  I first factored it into smaller chunks for RuboCop then pulled it into here for Pronto which complained (rightly so) about FeatureEnvy.